src/main/java/nl/pim16aap2/armoredElytra/util/EnchantmentContainer.java

```diff
@@ -185,7 +205,11 @@ public static EnchantmentContainer merge(final EnchantmentContainer first, final
         final Map<Enchantment, Integer> combined = new HashMap<>(first);
         for (Map.Entry<Enchantment, Integer> entry : second.entrySet())
         {
-            Integer enchantLevel = first.get(entry.getKey());
+            // Check for mutually exclusive enchantment (giving second entry priority)
+            combined.keySet().removeIf(firstEnchantment -> areMutuallyExclusive(firstEnchantment, entry.getKey()));
```

Comment:
I've implemented measures to limit calls into the nested loops in 28e30265c50420b7f39f1d1435bedcfd6bee8183
It will now check the first item for its enchants and add all of their mutually exclusive enchantments to a list, as well as add the scanned enchantment to an "allowed" list

Then when it checks the second item it only scans enchantments that are not already marked as allowed, and it only checks the mutually exclusive enchantments of enchantments that are not already marked as disallowed.

Reply:
I've also fixed a facepalm in 6435c825478ed4946fbd73768bcfab1d5a32db5c

Reply:
Hey, I really like the progress! However, there are two things I was wondering about:

1) It might be nice to filter out mutually exclusive values in the EnchantmentContainer constructor itself, right after [this](https://github.com/PimvanderLoos/ArmoredElytra/blob/1fe5bc2634a7d6d36c9e4b9ca80eccc80f024f50/src/main/java/nl/pim16aap2/armoredElytra/util/EnchantmentContainer.java#L25) filter call. This way we can ensure that there is no situation where mutually exclusive enchantments slip through. Currently, this could happen when taking the enchantments from an item (`EnchantmentContainer#getEnchantments`) and then applying it on an armored elytra, in which case the whole merge method is never invoked. Additionally, if the merge method is invoked and one of the containers is null or empty, the other container is returned, which could contain mutually exclusive enchantments.
If we filter out mutually exclusive enchantments in the ctor, we can assume that neither container contains any, so after building the disallowedEnchantments collection, we can use `Collection#removeAll(Collection)` to filter out the disallowed enchantments from the other set before the merge-loop. This has the advantage that we don't have to update the disallowedEnchantments collection in the loop anymore as any enchantments left in the second set are guaranteed to not be mutually exclusive.
2) Currently, the disallowedEnchantments collection is first created using the first item, meaning any items in the second set that are mutually exclusive with an item in the first set are discarded, while the second item should take precedence according to the current specifications. Ideally, the disallowedEnchantments collection should be constructed from the second container and be used to filter entries from the first container.

Reply:
In 5fb902d46f8de51c6ce97562caf85e73d1ed38ae I've moved the check into the constructor and made `EnchantmentContainer#getMutuallyExclusiveEnchantments(Enchantment)` private.
Since I am no longer putting the method inside of merge the second point is also no longer an issue.